server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -653,9 +528,10 @@ public TSExecuteStatementResp executeRawDataQuery(TSRawDataQueryReq req) {
    */
   @SuppressWarnings("squid:S3776") // Suppress high Cognitive Complexity warning
   private TSExecuteStatementResp internalExecuteQueryStatement(String statement,
-      long statementId, PhysicalPlan plan, int fetchSize, String username) throws IOException {
+      long statementId, PhysicalPlan plan, int fetchSize, String username)
+      throws QueryProcessException, SQLException, StorageEngineException, QueryFilterOptimizationException, MetadataException, IOException, InterruptedException, TException, AuthException {
```

Comment:
I'm not sure about this. Since lots of Exceptions are thrown here and adopt the same process outside, can we just catch and process them in this method?

Reply:
First, I think we should try to catch errors in the outermost call.
In addition, the error handling process of internalExecuteQueryStatement is roughly the same as that of its caller, and we should not let the similar error handling process be executed twice.